extensions/amp-video/0.1/amp-video.js

```diff
@@ -325,9 +325,16 @@ class AmpVideo extends AMP.BaseElement {
     }
 
     // loadPromise for media elements listens to `loadedmetadata`.
-    const promise = this.loadPromise(this.video_).then(() => {
-      this.element.dispatchCustomEvent(VideoEvents.LOAD);
-    });
+    const promise = this.loadPromise(this.video_).then(
+      () => {
+        this.element.dispatchCustomEvent(VideoEvents.LOAD);
+      },
+      () => {
+        // Ensures the load failure property is set on the right video element,
+        // as they can be swapped by the Stories media pool.
```

Comment:
Can you explain? I don't think amp-video should be responsible for ever setting `LOAD_FAILURE_PROPERTY`.

Reply:
I might a bit paranoid here, but given what I explained about the Stories media pool swapping video element, I'm worried about the following race condition:
1. `amp-video.layoutCallback` calls `loadPromise` with the original video
2. Stories media pool swaps the video for its own
3. `loadPromise` fails and sets `LOAD_FAILURE_PROPERTY` on the original video, that is no longer in the DOM

This line ensures that there's a 4rth step, where if the video got swapped between step (1) and (3), the error still gets applied to the right video element.
I haven't been able to repro that during local testing but I think it could happen in production.

Reply:
Ok. I don't think this is amp-video's responsibility then. It seems loadPromise should check that the media is still pointing at the same content before storing the error/success.

Reply:
Got it. Removed it